packages/node_modules/@webex/webex-core/test/unit/spec/interceptors/embargo.js

```diff
@@ -0,0 +1,139 @@
+        beforeEach('set appropriate status code and spys', () => {
+          reason = new WebexHttpError.InternalServerError({
+            message: 'test message',
+            statusCode: 451,
+            options: {
+              url: 'http://not-a-url.com/',
+              headers: {
+                trackingId: 'tid'
+              }
+            }
+          });
+        });
+
+        it('should return a rejected promise with the reason', () =>
+          assert.isRejected(interceptor.onResponseError(options, reason))
+            .then((error) => assert.equal(reason, error)));
+
+        it('should clear credentials', () =>
+          assert.isRejected(interceptor.onResponseError(options, reason))
+            .then(() => assert.called(credentialsClear)));
+
+        it('should present an appropriate logger message', () =>
+          assert.isRejected(interceptor.onResponseError(options, reason))
+            .then(() => assert.calledWith(loggerInfo, loggerMessage)));
+
+        describe('when the device plugin is mounted', () => {
+          let deviceClear;
+
+          beforeEach('set up the device plugin', () => {
+            interceptor.webex.internal.device = {
+              clear: sinon.spy()
+            };
+
+            deviceClear = interceptor.webex.internal.device.clear;
+          });
+
+          it('should clear the device', () =>
+            assert.isRejected(interceptor.onResponseError(options, reason))
+              .then(() => assert.called(deviceClear)));
+        });
+      });
+
+      describe('when the reason does not have a \'451\' status code', () => {
+        beforeEach('set appropriate status code and spys', () => {
+          reason = new WebexHttpError.InternalServerError({
+            message: 'test message',
+            statusCode: 452,
+            options: {
+              url: 'http://not-a-url.com/',
+              headers: {
+                trackingId: 'tid'
+              }
+            }
+          });
+        });
+
+        it('should return a rejected promise with the reason', () =>
+          assert.isRejected(interceptor.onResponseError(options, reason))
+            .then((error) => assert.equal(reason, error)));
+
+        it('should not clear credentials', () =>
+          assert.isRejected(interceptor.onResponseError(options, reason))
+            .then(() => assert.notCalled(credentialsClear)));
+
+        it('should not present any logger message', () =>
+          assert.isRejected(interceptor.onResponseError(options, reason))
+            .then(() => assert.notCalled(loggerInfo)));
+
+        describe('when the device plugin is mounted', () => {
+          let deviceClear;
+
+          beforeEach('set up the device plugin', () => {
+            interceptor.webex.internal.device = {
+              clear: sinon.spy()
+            };
+
+            deviceClear = interceptor.webex.internal.device.clear;
+          });
+
+          it('should not clear the device', () =>
```

Comment:
Shouldn't this be true according to implementation code:

```js
      if (device) {
        this.webex.internal.device.clear();
      }
```

Reply:
If a user is utilizing `webex-core` without the device plugin being mounted, it will fail here since `clear()` will not exists on `undefined` `device`. Previously, this interceptor was mounted with `device`, so that the check wasn't required.

Reply:
Oh, I'm sorry, I misread the describe blocks. Was thinking this was for the 451 error.